Core/include/Acts/Utilities/BinAdjustment.hpp

```diff
@@ -240,26 +240,28 @@ static inline BinUtility adjustBinUtility(const BinUtility& bu,
                                           const Surface& surface,
                                           const GeometryContext& gctx) {
   // The surface type is a cylinder
-  if (surface.type() == Surface::Cylinder) {
+  if (surface.type() == Surface::Cylinder &&
+      surface.bounds().type() == SurfaceBounds::eCylinder) {
     // Cast to Cylinder bounds and return
-    auto cBounds = dynamic_cast<const CylinderBounds*>(&(surface.bounds()));
+    auto cBounds = static_cast<const CylinderBounds*>(&(surface.bounds()));
```

Comment:
is that safe with virtual inheritance? or only as long as we do no diamond stuff?